### PR TITLE
Wd 38 woo commerce create ui test for credit card config

### PR DESCRIPTION
**Pull request by ivanradakovic987**

### CODE REVIEW

File: tests/_support/Step/Acceptance/ShopSystem/WoocommerceStep.php [hunk 1]
```diff
@@ -349,4 +352,143 @@ public function putShippingZoneInDatabase($zoneName, $zoneRegions, $shippingMeth
...
+        $paymentMethodPageLocator  = strtolower($paymentMethod) . '_payment';
+        foreach ($paymentMethodConfig as $name => $value) {
+            // check the type of element based on name of locator
+            if (array_key_exists($name . '_text', $this->getLocator()->$paymentMethodPageLocator)) {
+                $locator = $name . '_text';
+                $this->preparedFillField($this->getLocator()->$paymentMethodPageLocator->$locator, $value);
+            } elseif (array_key_exists($name.'_select', $this->getLocator()->$paymentMethodPageLocator)) {
+                $locator = $name . '_select';
+                //payment action should be taken from parameter
+                if ($name == 'payment_action') {
```
[the remark this answers is not in the document]
Comment (ivanradakovic987) [reply tests/_support/Step/Acceptance/ShopSystem/WoocommerceStep.php:429]: Codacy doesn't allow else clauses, it considers them unnecessary. I'll try to figure out a way to do this.

File: tests/_support/Step/Acceptance/ShopSystem/WoocommerceStep.php [hunk 2]
```diff
@@ -349,4 +352,143 @@ public function putShippingZoneInDatabase($zoneName, $zoneRegions, $shippingMeth
             [static::SHIPPING_ZONE_ID_COLUMN_NAME => $zoneId]
         );
     }
+
+    /**
+     * @throws Exception
+     */
+    public function logInToAdministrationPanel()
```
[the remark this answers is not in the document]
Comment (ivanradakovic987) [reply tests/_support/Step/Acceptance/ShopSystem/WoocommerceStep.php:359]: Good proposal. Should we create user story for this task?

[the remark this answers is not in the document]
Comment (tatsta) [reply tests/_support/Step/Acceptance/ShopSystem/WoocommerceStep.php:359]: Yes, we can do that in a separate US. Would you inform PO about this?

File: tests/_support/Step/Acceptance/ShopSystem/WoocommerceStep.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (ivanradakovic987) [reply tests/_support/Step/Acceptance/ShopSystem/WoocommerceStep.php]: Done.

File: tests/_support/Step/Acceptance/ShopSystem/WoocommerceStep.php (hunk 2 above)
[the remark this answers is not in the document]
Comment (ivanradakovic987) [reply tests/_support/Step/Acceptance/ShopSystem/WoocommerceStep.php:361]: Yes, no prob.

File: tests/_support/Step/Acceptance/ShopSystem/WoocommerceStep.php [hunk 3]
```diff
@@ -349,4 +352,143 @@ public function putShippingZoneInDatabase($zoneName, $zoneRegions, $shippingMeth
...
+
+        $paymentMethodPageLocator = strtolower($paymentMethod) . '_payment';
+        foreach ($this->paymentMethodConfig as $name => $value) {
+            // check the type of element based on name of locator
+            if (array_key_exists($name . '_text', $this->getLocator()->$paymentMethodPageLocator)) {
+                $locator = $name . '_text';
+                $this->seeInField($this->getLocator()->$paymentMethodPageLocator->$locator, $value);
+            } elseif (array_key_exists($name.'_select', $this->getLocator()->$paymentMethodPageLocator)) {
+                $locator = $name . '_select';
+                if ($name == 'payment_action') {
```
[the remark this answers is not in the document]
Comment (ivanradakovic987) [reply tests/_support/Step/Acceptance/ShopSystem/WoocommerceStep.php]: Done.

File: tests/acceptance/CreditCard/CreditCardConfigurationHappyPath.feature [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+  As admin user
+  I want to check that the credit card configuration page is working properly
+
+  Background:
+    Given I initialize shop system
+
+  @woocommerce @test
+  Scenario Outline: initial transaction Non 3DS
+    Given I deactivate "CreditCard" payment method in configuration
+    When I go into the configuration mask as "admin user" and activate "CreditCard" method
```
[the remark this answers is not in the document]
Comment (ivanradakovic987) [reply tests/acceptance/CreditCard/CreditCardConfigurationHappyPath.feature]: I've created feature file based on user story. You're right, though, so I replaced mask with page. :-)

File: tests/acceptance/CreditCard/CreditCardConfigurationHappyPath.feature [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+    And I fill fields with "CreditCard" data for payment action <payment_action> and transaction type <transaction_type>
+    #And I go to Payment page and check that "CreditCard" payment method is enabled
```
[the remark this answers is not in the document]
Comment (ivanradakovic987) [reply tests/acceptance/CreditCard/CreditCardConfigurationHappyPath.feature]: Done.

**Pull request merged**